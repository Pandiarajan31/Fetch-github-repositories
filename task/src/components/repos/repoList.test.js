import React from 'react';
import ApolloClient, { InMemoryCache } from 'apollo-boost';
import RepoList from './repoList';
import renderer from 'react-test-renderer'
import { ApolloProvider } from 'react-apollo';

// renders correctly when there is props

it('renders correctly when there is props', () => {
    const toggleState = "starred";
    const inputSearch = "test";
    const token = jest.fn(() => "208a70fda62f543564616b2aa57b8bcc9b40cf98");

    const client = new ApolloClient({
        uri: 'https://api.github.com/graphql',
        cache: new InMemoryCache(),
        request: operation => {
          operation.setContext({
            headers: {
              authorization: `Bearer ${token}`
            }
          });
        }
      });
    
    const tree = renderer.create(
        <ApolloProvider client={ client } >

    <RepoList toggleState={toggleState} inputSearch={inputSearch} changeToggle={()=>console.log("done")}  />
    </ApolloProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

