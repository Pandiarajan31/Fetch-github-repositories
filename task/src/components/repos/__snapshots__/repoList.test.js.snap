// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is props 1`] = `
<div
  className="repo-list"
  data-radium={true}
>
  <div
    data-radium={true}
    style={
      Object {
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    <img
      alt="loader"
      src="loading.gif"
    />
  </div>
</div>
`;
